client/gutenberg/editor/controller.js

```diff
@@ -109,7 +109,7 @@ function waitForPreferredEditorView( context ) {
  *
  * @param {object} context  Shared context in the route.
  * @param {Function} next   Next registered callback for the route.
- * @returns {*}             Whatever the next callback returns.
+ * @returns {undefined}             Whatever the next callback returns.
```

Comment:
According to its TS definition, the callback can return any value:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/page/index.d.ts#L257

Therefore changing this to `undefined` could cause an inconsistency. 

We can enable the any type in JSDoc by delcaring `*` in `settings.jsdoc.preferredTypes`.

Reply:
👍

Reply:
It looks like this triggers a bug in eslint jsdoc, where `*` by itself doesn't handle a void/undefined return type. So we have to write `*|undefined`, which I guess is fine for now

Reply:
Bug report: https://github.com/gajus/eslint-plugin-jsdoc/issues/950